Close #71: Add sha256_stub - RAM loaded firmware to calculate SHA256 hash of flashed firmware

TheSomeMan opened the pull request.

TheSomeMan committed:
Close #71: Add sha256_stub - RAM loaded firmware to calculate SHA256 hash of flashed firmware

ojousima reviewed (approved):
Beautiful implementation for verifying flash overall, great work. 

Has the SHA256 algorithm been verified against any public test vectors? If not, would it be simple enough to add tests, as in less than a day of effort?

TheSomeMan committed:
Add unit tests for sha256_stub

Wrap `stub_reset` and `stub_entry` functions with `#ifndef CEEDLING`
to prevent conflicts during unit testing with Ceedling.

The new tests verify the SHA256 calculation against FIPS 180-4
standard test vectors for empty, single-block, and multi-block messages.

TheSomeMan committed:
Merge branch 'master' into issue-71

TheSomeMan committed:
docs(sha256_stub): #71 Add detailed comments for flash hashing stub functions and parameters
